Index: fstools-2016-07-24/block.c
===================================================================
--- fstools-2016-07-24.orig/block.c	2016-08-02 22:45:13.001837208 +0800
+++ fstools-2016-07-24/block.c	2016-08-02 22:48:13.629838081 +0800
@@ -518,17 +518,19 @@
 
 static void cache_load(int mtd)
 {
-	if (mtd) {
-		_cache_load("/dev/mtdblock*");
-		_cache_load("/dev/ubiblock*");
-		_cache_load("/dev/ubi[0-9]*");
-	}
 	_cache_load("/dev/mmcblk*");
 	_cache_load("/dev/sd*");
 	_cache_load("/dev/hd*");
 	_cache_load("/dev/md*");
 	_cache_load("/dev/vd*");
 	_cache_load("/dev/mapper/*");
+
+
+       if (mtd) {
+               _cache_load("/dev/mtdblock*");
+               _cache_load("/dev/ubiblock*");
+               _cache_load("/dev/ubi?*_?*");
+       }
 }
 
 static int print_block_info(struct blkid_struct_probe *pr)
